### PR TITLE
refactor: remove legacy java frontend

### PR DESCRIPTION
**Pull request by fuyufjh**
## What's changed and what's your intention?

Remove legacy Java frontend, including sources and tests.

***TO REVIEWERS: Click this link to see diffs excluding removing legacy folder*** ->  https://github.com/singularity-data/risingwave/pull/2519/files/2ef98f693fb42eeb1bf0df3facfa95549f4b35f2..HEAD

Some remaining works need to be done later:

- Check E2E test cases and make sure all cases are covered in `v2/` before removing them
- Migrate regress tests `src/tests/regress/` to Rust frontend and enable [them](https://github.com/singularity-data/risingwave/pull/2519/files/2ef98f693fb42eeb1bf0df3facfa95549f4b35f2..bc070640072b16abd5f4afb276eb875dce97970a#diff-de94810cb4bba3cd527324abf6f643f708c92396756ea14c50de470beb643f22R118)
- Remove compatible code in backend (keyword: `is_legacy_frontend`)
- Remove `CreateSourceExecutor`
- Remove `create_materialized_view` and `drop_materialized_view` in `StreamServiceImpl` #2260
- Rename all occurrences of "FrontendV2" to "Frontend"

## Checklist

N/A

## Refer to a related PR or issue link (optional)

None

### CODE REVIEW

File: risedev.yml [hunk 1]
```diff
@@ -250,7 +227,7 @@ risedev:
     - use: compute-node
       port: 5689
       exporter-port: 1224
-    - use: frontend-legacy
+    - use: frontend
```
Comment (skyzh) [inline risedev.yml:230]: ci-3node is replaced by `ci-3cn-1fe`. This configuration won't be used in CI, and can be removed.

**Review by skyzh — APPROVED** [review]
Cool!

**Comment by D2Lark** [comment]
some meta service used by legacy java frontend can be removed~ like [this](https://github.com/singularity-data/risingwave/issues/2260)

**Comment by TennyZhuang** [comment]
> some meta service used by legacy java frontend can be removed~ like [this](https://github.com/singularity-data/risingwave/issues/2260)

For review, We can do it in another PR!

**Comment by TennyZhuang** [comment]
How about tag a new version?

**Comment by fuyufjh** [comment]
> How about tag a new version?

Good idea! Just pushed a new tag `v0.1.7`. The release pipeline has been triggered. https://github.com/singularity-data/risingwave/actions/runs/2324374089

**Comment by fuyufjh** [comment]
> some meta service used by legacy java frontend can be removed~ like [this](https://github.com/singularity-data/risingwave/issues/2260)

Added to the todo list in PR description. Let's open a tracking issue and do these later.

**Pull request merged**